app/Services/EntityFileService.php

```diff
@@ -16,33 +20,56 @@ class EntityFileService
     /**
      * @throws EntityFileException
      */
-    public function upload(StoreEntityAsset $request, string $field = 'file', string $folder = 'entities/files'): EntityAsset
+    public function upload(StoreEntityAsset $request, string $field = 'file'): EntityAsset
     {
+        /** @var GalleryService $service */
+        $service = app()->make(GalleryService::class);
+
+        // Prepare the file for the journey
+        $uploadedFile = $request->file($field);
+
         // Already above max capacity?
-        if ($this->entity->files->count() >= $this->campaign->maxEntityFiles()) {
+        if ($this->entity->files->count() >= $this->campaign->maxEntityFiles() || $service->campaign($this->campaign)->available() < $uploadedFile->getSize() / 1024) {
```

Comment:
The exception if the storage is full should be different and clearly explained to the user